commit 3bfda62c50b0a4b118dcfce36686508ca2892292
Author: Will Hawkins <hawkinsw@opentechinstitute.org>
Date:   Tue Jun 19 17:59:14 2012 -0400

    mac80211: Allow userspace to register for auth frames in IBSS
    
    Set the necessary flags to allow user space applications
    to register for authentication frames on IBSS interfaces.
    This is useful for situations where userspace applications
    want to control key negotiation between stations.
    
    Signed-off-by: Will Hawkins <hawkinsw@opentechinstitute.org>
    [reword commit message a bit]
    Signed-off-by: Johannes Berg <johannes.berg@intel.com>

diff --git a/net/mac80211/main.c b/net/mac80211/main.c
index 976e41365c25..0b040fb73673 100644
--- a/net/mac80211/main.c
+++ b/net/mac80211/main.c
@@ -462,7 +462,9 @@ static const struct ieee80211_txrx_stypes
 ieee80211_default_mgmt_stypes[NUM_NL80211_IFTYPES] = {
 	[NL80211_IFTYPE_ADHOC] = {
 		.tx = 0xffff,
-		.rx = BIT(IEEE80211_STYPE_ACTION >> 4),
+		.rx = BIT(IEEE80211_STYPE_ACTION >> 4) |
+			BIT(IEEE80211_STYPE_AUTH >> 4) |
+			BIT(IEEE80211_STYPE_DEAUTH >> 4),
 	},
 	[NL80211_IFTYPE_STATION] = {
 		.tx = 0xffff,